sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -323,11 +389,24 @@ case class Add(
 
   override def decimalMethod: String = "$plus"
 
+  override def decimalType(p1: Int, s1: Int, p2: Int, s2: Int): DecimalType = {
```

Comment:
we should copy the doc from `DecimalPrecision` to here for the ADD opration.

Reply:
```
 *   Operation    Result Precision                        Result Scale
 *   ------------------------------------------------------------------------
 *   e1 + e2      max(s1, s2) + max(p1-s1, p2-s2) + 1     max(s1, s2)
 *   e1 - e2      max(s1, s2) + max(p1-s1, p2-s2) + 1     max(s1, s2)
 *   e1 * e2      p1 + p2 + 1                             s1 + s2
 *   e1 / e2      p1 - s1 + s2 + max(6, s1 + p2 + 1)      max(6, s1 + p2 + 1)
 *   e1 % e2      min(p1-s1, p2-s2) + max(s1, s2)         max(s1, s2)
 *   e1 union e2  max(s1, s2) + max(p1-s1, p2-s2)         max(s1, s2)
```

do you mean we should copy this docs into each operator ?

Reply:
only the `e1 + e2` one.

Reply:
added